hdf5-sys/build.rs

```diff
@@ -584,27 +593,123 @@ impl Config {
         vs.extend((0..=5).map(|v| Version::new(1, 10, v))); // 1.10.[0-5]
         for v in vs.into_iter().filter(|&v| version >= v) {
             println!("cargo:rustc-cfg=hdf5_{}_{}_{}", v.major, v.minor, v.micro);
+            println!("cargo:version_{}_{}_{}=1", v.major, v.minor, v.micro);
         }
         if self.header.have_stdbool_h {
             println!("cargo:rustc-cfg=h5_have_stdbool_h");
+            println!("cargo:have_stdbool=1");
         }
         if self.header.have_direct {
             println!("cargo:rustc-cfg=h5_have_direct");
+            println!("cargo:have_direct=1");
         }
         if self.header.have_parallel {
             println!("cargo:rustc-cfg=h5_have_parallel");
+            println!("cargo:have_parallel=1");
         }
         if self.header.have_threadsafe {
             println!("cargo:rustc-cfg=h5_have_threadsafe");
+            println!("cargo:have_threadsafe=1");
         }
     }
 }
 
 fn main() {
-    let mut searcher = LibrarySearcher::new_from_env();
-    searcher.try_locate_hdf5_library();
-    let config = searcher.finalize();
-    println!("{:#?}", config);
-    config.emit_link_flags();
+    if feature!("STATIC").is_ok() {
+        build_and_emit();
+    } else {
+        let mut searcher = LibrarySearcher::new_from_env();
+        searcher.try_locate_hdf5_library();
+        let config = searcher.finalize();
+        println!("{:#?}", config);
+        config.emit_link_flags();
+        config.emit_cfg_flags();
+    }
+}
+
+fn build_and_emit() {
+    println!("cargo:rerun-if-changed=build.rs");
+    let mut hdf5config = cmake::Config::new("source");
+    hdf5config
+        .define("HDF5_NO_PACKAGES", "ON")
+        .define("BUILD_SHARED_LIBS", "OFF") // Only need to build a static library
+        .define("BUILD_TESTING", "OFF")
+        .define("HDF5_BUILD_TOOLS", "OFF")
+        .define("HDF5_BUILD_EXAMPLES", "OFF")
+        //
+        .define("HDF5_BUILD_JAVA", "OFF")
+        .define("HDF5_BUILD_FORTRAN", "OFF")
+        .define("HDF5_BUILD_CPP_LIB", "OFF")
+        // Default, with feature flags
+        .define("HDF5_ENABLE_PARALLEL", "OFF")
+        .define("HDF5_ENABLE_DEPRECATED_SYMBOLS", "OFF")
+        .define("HDF5_ENABLE_THREADSAFE", "OFF")
+        .define("ALLOW_UNSUPPORTED", "OFF")
+        .define("HDF5_BUILD_HL_LIB", "OFF");
+
+    if feature!("ZLIB").is_ok() {
+        let zlib_include_dir = std::env::var_os("DEP_Z_INCLUDE").unwrap();
+        let mut zlib_header = std::env::split_paths(&zlib_include_dir).nth(0).unwrap();
+        zlib_header.push("zlib.h");
```

Comment:
Wonder what to do about zlib, if we're packaging 20mb of hdf5 source, shouldn't we just package zlib as well? I.e., when would the user choose to point at their own zlib.h?

Reply:
Ah, apologies, I've misread this; does DEPS_Z_INCLUDE come from libz-sys/static? So it is self-sufficient then.

Reply:
Yeah, `DEP_{LIB}_{SOMETHING}` is included from the dependency. This way of passing flags up is enabled through `links` in `Cargo.toml`